taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -468,53 +471,83 @@ func (self *Contract) Run(ctx vm.CallFrame, evm *vm.EVM) ([]byte, error) {
 			fmt.Println("Unable to parse getValidators input args: ", err)
 			return nil, err
 		}
-
-		res := self.getValidators(args)
-		return method.Outputs.Pack(res.Validators, res.End)
+		return method.Outputs.Pack(self.getValidators(args))
 
 	case "getDelegations":
 		var args sol.GetDelegationsArgs
 		if err = method.Inputs.Unpack(&args, input); err != nil {
 			fmt.Println("Unable to parse getDelegations input args: ", err)
 			return nil, err
 		}
-
-		res := self.getDelegations(args)
-		return method.Outputs.Pack(res.Delegations, res.End)
+		return method.Outputs.Pack(self.getDelegations(args))
 
 	case "getUndelegations":
 		var args sol.GetUndelegationsArgs
 		if err = method.Inputs.Unpack(&args, input); err != nil {
 			fmt.Println("Unable to parse getUndelegations input args: ", err)
 			return nil, err
 		}
-
-		res := self.getUndelegations(args)
-		return method.Outputs.Pack(res.Undelegations, res.End)
+		return method.Outputs.Pack(self.getUndelegations(args))
 	}
 
 	return nil, nil
 }
 
-func (self *Contract) DistributeRewards(rewardsStats *rewards_stats.RewardsStats, feesRewards *FeesRewards) {
+func (self *Contract) DistributeRewards(author *common.Address, rewardsStats *rewards_stats.RewardsStats, feesRewards *FeesRewards) {
```

Comment:
Can you add rewards distribution algo description ? Everyone, who will either do this review, or will adjust the code in the future is going to need it